added and customized Book Controller,

JaysonJuza opened the pull request:
added BookDto and CreateBookRequestDto objects for the Book entity and for use at the service level, added the mapstruct library and added the corresponding dependency to pom.xml, added MapperConfig configuration class ,
added BookMapper for converting entity to DTO object and vice versa, custom exception added

JaysonJuza committed:
added and customized Book Controller,
added BookDto and CreateBookRequestDto objects for the Book entity and for use at the service level,
added the mapstruct library and added the corresponding dependency to pom.xml,
added MapperConfig configuration class ,
added BookMapper for converting entity to DTO object and vice versa,
custom exception added

MAMentorFX commented on src/main/java/onlinebookstoreapp/controller/BookController.java:
```diff
@@ -0,0 +1,39 @@
+    @Autowired
+    public BookController(BookService bookService) {
+        this.bookService = bookService;
+    }
```
Optional: consider @RequiredArgsConstructor.

MAMentorFX commented on src/main/java/onlinebookstoreapp/dto/BookDto.java:
```diff
@@ -0,0 +1,17 @@
+import lombok.Setter;
+
+@Getter
+@Setter
```
It's better to use @Data for Dtos

MAMentorFX commented on src/main/java/onlinebookstoreapp/mapper/BookMapper.java:
```diff
@@ -0,0 +1,14 @@
+public interface BookMapper {
+    BookDto toDtoModel(Book book);
+
+    Book toBookModel(CreateBookRequestDto requestDto);
```
Either toBookDto + toBook, or toDto + toModel

MAMentorFX commented on src/main/java/onlinebookstoreapp/service/impl/BookServiceImpl.java:
```diff
@@ -10,19 +14,31 @@
+    public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
         this.bookRepository = bookRepository;
+        this.bookMapper = bookMapper;
     }
```
Speaking of @RequiredArgsConstructor: that's why you wont be bothered changing constructor in the future, when you'll be adding more and more fields.

MAMentorFX commented on src/main/java/onlinebookstoreapp/service/impl/BookServiceImpl.java:
```diff
@@ -10,19 +14,31 @@
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDtoModel)
+                .toList();
```
MapStruct supports mapping lists

JaysonJuza committed:
Replaced Beans Factory @Autowired annotation to Lombok annotation @RequiredArgsConstructor at the repository, service and controller layers.
Replaced Lombok annotations @Getter, @Setter to @Data in the BookDto.class and CreateBookRequestDto.class.
Changes mapping methods name.
Added mapping methods toListDto().
